android/src/main/java/com/mapbox/mapboxgl/setMapLanguage.kt

```diff
@@ -0,0 +1,30 @@
+@file:JvmName("MapboxMapUtils")
+
+package com.mapbox.mapboxgl
+
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.style.expressions.Expression
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory
+import com.mapbox.mapboxsdk.style.layers.SymbolLayer
+
+fun MapboxMap.setMapLanguage(language: String) {
+    val layers = this.style?.layers ?: emptyList()
+
+    val languageRegex = Regex("(name:[a-z]+)")
```

Comment:
I assume the idea here is to skip any layers with textfields that do not depend on the name from the tiles. But is also on purpose to ignore layers with` text = ["get", "name"] `?

Reply:
We already ignore layers where `["get", "name"]` is present, since the layer check for at least a `: `.

Reply:
Yeah, that's my question: is that on purpose?

Reply:
Not saying it's right or wrong, just occurred to me that I'm not sure whether a developer would expect such a layer to be localized or not...

Reply:
Thinking about it more, i think there is a good argument for either approach:
1. On the one hand, if the style author sets text = ["get", "name"] they may explicitly **always** want to show the non-translated name here --> we should ignore these layers
2. On the other hand, if the developer explicitly calls this method to localize map labels, they might expect **all** map labels to be localized --> these layers should be localized as well

Reply:
I would definitely go with option because of multiple reasons: 

1. If the author just set `["get", "name"]` or `[ref]`, they want the plain name for sure.
2. The developer should not care about which layer will be translated and which not, since the most developer are not familiar about the different layers and how the styling is working in detail.
3. The `setMapLanguage` function would not work in our case, since we have a lot of usages of `["get", "name"]` and `[ref]`, and the text is just empty in that cases.